### ISSUE TITLE
PRIORITY 1: Internal server error - attempt to edit client energy record - and pull back rates/total from API to client tariff records

### ISSUE THREAD

**Issue by iddmaxwell** [iddmaxwell/optimisebills#85]
Server Error in '/' Application.

The remote server returned an error: (500) Internal Server Error. 
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.Net.WebException: The remote server returned an error: (500) Internal Server Error.

Source Error: 

Line 396:            var elecTariffUrl = "http://services.ukpower.co.uk/home_energy/api/tariffs.xml?api_key=8B34-7EDA-E529-B25D-8755&limit=0" + tariffQuery;
Line 397:
Line 398:            elecTariffDoc = XDocument.Load(elecTariffUrl);
Line 399:            var topTariffList = elecTariffDoc.Descendants("tariffs").ToList().FirstOrDefault();
Line 400:            var tariffs = topTariffList.Descendants("tariff").ToList();

Source File: c:\inetpub\wwwroot\Admin\EnergyRecordEditor.aspx.cs    Line: 398 

Stack Trace: 

[WebException: The remote server returned an error: (500) Internal Server Error.]
   System.Net.HttpWebRequest.GetResponse() +1126
   System.Xml.XmlDownloadManager.GetNonFileStream(Uri uri, ICredentials credentials) +86
   System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn) +65
   System.Xml.XmlReader.Create(String inputUri, XmlReaderSettings settings, XmlParserContext inputContext) +200
   System.Xml.Linq.XDocument.Load(String uri, LoadOptions options) +48
   EnergyRecordManager.GasDropDownChange(DropDownList supplier, DropDownList NewTariff, DropDownList SType, DropDownList Area) in c:\inetpub\wwwroot\Admin\EnergyRecordEditor.aspx.cs:398
   EnergyRecordManager.Page_Load(Object sender, EventArgs e) in c:\inetpub\wwwroot\Admin\EnergyRecordEditor.aspx.cs:61
   System.Web.Util.CalliHelper.EventArgFunctionCaller(IntPtr fp, Object o, Object t, EventArgs e) +25
   System.Web.Util.CalliEventHandlerDelegateProxy.Callback(Object sender, EventArgs e) +42
   System.Web.UI.Control.OnLoad(EventArgs e) +132
   System.Web.UI.Control.LoadRecursive() +11142047
   System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) +2428

Version Information: Microsoft .NET Framework Version:2.0.50727.5485; ASP.NET Version:2.0.50727.5491

**Comment by iddmaxwell** [comment]
Looks similar to issue 77 which was closed just over a week ago. i'm not sure it is fixed

**Comment by Will-Newton** [comment]
Pull back total per annum from api on energy record editor

**Comment by iddmaxwell** [comment]
Looks like the rates are being stored correctly on the energy record when I edit it!  

However, I got an internal server error 500 on attempting to create a new comparison record and update tariff for Delphine Park.  I couldn't find her actual EON tariff so created a BG comparison record intending to edit but get the error.

**Comment by Will-Newton** [comment]
Which page is this on?

**Comment by iddmaxwell** [comment]
Go to Delphine Park.
Click View/Edit on the British Gas comparison record.

**Comment by Will-Newton** [comment]
Fixed

**Comment by iddmaxwell** [comment]
Issue with Delphine Park is fixed but the tariff editor is no longer pulling back rates via the API.
See Jean Maxwell, edit current tariff.  I updated the kwh figures and no rates are on the record so can't recalculate. However I then changed the tariff to force it to pull back rates but none have appeared. 
I thought this was working yesterday!

**Comment by Will-Newton** [comment]
I don't think it was. I think there's a problem that the rates are on the quotes feed which is separate from the tariff feed (used here to pull back tariffs for the drop down list)
